app/routes/admin.settings.tsx

```diff
@@ -312,9 +312,24 @@ export const action = async ({ request }: ActionFunctionArgs) => {
       } else if (body.get('intent') === 'delete-email-template') {
         await StoreConfig.deleteEmailTemplateByName(String(body.get('name')))
         return json({ error: null, data: {} }) //for modal dismissal
+      } else if (body.get('intent') === 'create-category') {
+        await new ProductCategoryModel().create({
+          id: '',
+          name: String(body.get('name')),
+          slug: String(body.get('slug')),
+        })
+      } else if (body.get('intent') === 'update-category') {
+        await new ProductCategoryModel().update({
+          id: payload.id,
+          name: String(body.get('name')),
+          slug: String(body.get('slug')),
+          parentId: String(body.get('parentId')),
+        })
+        return json({ error: null, data: {} }) //for modal dismissal
+      } else if (body.get('intent') === 'delete-category') {
+        await new ProductCategoryModel().delete(payload.id)
+        return json({ error: null, data: {} })
       }
-
-      return json({ error: null, data: {} })
```

Comment:
hmm... you probably still need this line, TypeScript may hint you that you haven't covered all possible cases. coz theoretically, there might be some cases are not covered by all the if..else if cases